bindings/ch/_sir_strremove.chf

```diff
@@ -1,13 +1,17 @@
 /*
- * SPDX-License-Identifier: MIT
- *
  * _sir_strremove.chf
  *
  * Author:    Jeffrey H. Johnson <trnsz@pobox.com>
  * Copyright: Copyright (c) 2018-2023
  * Version:   2.2.4
  * License:   The MIT License (MIT)
  *
+ * -----------------------------------------------------------------------------
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2018-2023 Jeffrey H. Johnson <trnsz@pobox.com>
```

Comment:
It seems weird to duplicate this line essentially. There must be a way to combine them, even in the doxygen files

Reply:
Unfortunately I don't think there is.

If you mean the copyright line, it has to be that way to be recognized as valid license text, because it's part of the license, so it needs to be in the following format:
![IMG_8414](https://github.com/aremmell/libsir/assets/61629094/9eaea3d7-69e3-4a5c-9d0b-77bb239aa907)

SPDX and REUSE *do* allow you to completely get rid of the license text verbatim in each file, and use only a

If you want to do that, you then only need:

      SPDX-License-Identifier: MIT
      SPDX-FileCopyrightText: blah blah

If you don't use the full license text with a Copyright (c) .... then you should use the tag format, but if you have the full text it's not needed.  

As an aside, SPDX also allows you to distinguish between authors and other minor contributors too:
      SPDX-FileContributor: blah 
... which is where you'd list yourself if you took another file and modified it.

There are two "problems" though.  You cannot remove any license text or copyright notices that are already in the files from others (like the install script or those OpenBSD files we have).  So I decided to be consistent and have in every file three 'sections':

a) a prologue section which has the filename and possibly a description which may or may not be done with Doxygen tags.

b) any SPDX tags

c) the human readable license text.


---

REUSE/SPDX doesn't require a or c, but it does need it's specific format for b.

Reply:
Really, it would be nice if Doxygen supported SPDX and REUSE, but it doesn't seem to.

I looked at another project I use (RTEMS) that uses both REUSE and Doxygen, and they just don't use the license or copyright parts of Doxygen:

![IMG_8415](https://github.com/aremmell/libsir/assets/61629094/b974b409-41a8-45a6-a625-e09b6c45c826)